publiq-ui/pub-modal.stories.mdx

```diff
@@ -0,0 +1,100 @@
+import {
+  Meta,
+  Story,
+  Canvas,
+  Props,
+  ArgsTable,
+} from '@storybook/addon-docs/blocks';
+import PubModal, { ModalVariant } from './pub-modal.vue';
+import PubButton from '@/publiq-ui/pub-button';
+
+<Meta title="pub-modal" component={PubModal} />
+
+export const Template = (args, { argTypes }) => {
+  return {
+    props: Object.keys(argTypes),
+    data() {
+      return {
+        isVisible: false,
+      };
+    },
+    components: { PubModal, PubButton },
+    methods: {
+      handleClick() {
+        this.isVisible = true;
+      },
+      handleHide() {
+        this.isVisible = false;
+      },
+    },
+    template:
+      '<div><pub-button @click="handleClick">Open modal</pub-button><pub-modal v-bind="$props" :visible="isVisible" @hide="handleHide">Modal content</pub-modal></div>',
+  };
+};
+
+export const commonArgTypes = {
+  variant: {
+    control: {
+      type: false,
+    },
+  },
+};
+
+**
+This component is a wrapper around the [b-modal](https://bootstrap-vue.org/docs/components/modal) component from bootstrap-vue.
+You can use all of the defined methods and props from the bootstrap component.
```

Comment:
IMHO this allows way too much freedom and defeats the purpose of wrapping it with our own component? For example you can pass a lot of options that influence the styling of the modal: https://bootstrap-vue.org/docs/components/modal

It also makes it hard to decouple from this implementation later if we ever want to do a redesign, maybe using another component library underneath. I know that's an "if" at this point, but Corneel is open to that idea (but in the long run).

Reply:
I just don't want to end up in a situation where we need to keep our props in sync with the b-bootstrap ones. But let's discuss it in a meeting later, see the frontend channel

Reply:
Maybe there is a way to do it backwards? ignore style-specific props but pass all the other ones?